### PR TITLE
Drop Mesa workaround

### PR DESCRIPTION
**Pull request by rhaschke**
Addressing #1701. Needs to be tested on various systems!

### CODE REVIEW

File: src/rviz/ogre_helpers/render_system.cpp [hunk 1]
```diff
@@ -184,10 +183,6 @@ void RenderSystem::detectGlVersion()
     int major = caps->getDriverVersion().major;
     int minor = caps->getDriverVersion().minor;
     gl_version_ = major * 100 + minor * 10;
-
-    std::string gl_version_string = (const char*)glGetString(GL_VERSION);
-    // The "Mesa 2" string is intended to match "Mesa 20.", "Mesa 21." and so on
-    mesa_workaround = gl_version_string.find("Mesa 2") != std::string::npos && gl_version_ >= 320;
```
Comment (simonschmeisser) [inline src/rviz/ogre_helpers/render_system.cpp:190]: My suggestion would have been to just change this to "Mesa 20"

**Comment by simonschmeisser** [comment]
I was just testing around with the PC2 display on AMD Renoir  and noticed that most of the more fancy Styles seem to be unsupported with GLSL 1.5. Changing the Size has no effect for anything besides Points and it's always displayed as super tiny dots. I was not able to trigger any unusual behavior with the Points style

We have serious performance regressions with high resolution octomaps when limiting to the older shader versions though, that's why I opened the discussion.


```
[ INFO] [1644874202.850198482]: rviz version 1.14.14
[ INFO] [1644874202.850307278]: compiled against Qt version 5.12.8
[ INFO] [1644874202.850323799]: compiled against OGRE version 1.9.0 (Ghadamon)
[ INFO] [1644874202.865892892]: Forcing OpenGl version 0.
[ INFO] [1644874203.187769729]: Stereo is NOT SUPPORTED
[ INFO] [1644874203.187885107]: OpenGL device: AMD RENOIR (DRM 3.41.0, 5.13.0-28-generic, LLVM 12.0.0)
[ INFO] [1644874203.187928259]: OpenGl version: 4,6 (GLSL 4,6).
```

File: src/rviz/ogre_helpers/render_system.cpp (hunk 1 above)
Comment (simonschmeisser) [reply src/rviz/ogre_helpers/render_system.cpp:190]: Was the Mesa 20 part of some HWE or part of the stock 18.04, ie are there likely going to be people with non HWE systems running into this again?

**Comment by peci1** [comment]
Interesting, I also tested that on Renoir and it did not work for me with Mesa 21, that's why I requested the workaround to be extended to all 2x versions.

https://github.com/ros-visualization/rviz/pull/1598

**Comment by simonschmeisser** [comment]
Your test was with Mesa 21.0.0 while the fix appeared in 21.0.3 (I'm not sure it has been pinpointed exactly but maybe I misunderstood something ...)

**Comment by peci1** [comment]
@simonschmeisser Good point, haven't noticed that... Could you put together a minimum working example? It's been some time since I was solving this, so I can't remember what were the steps to receive a failure...

File: src/rviz/ogre_helpers/render_system.cpp (hunk 1 above)
Comment (peci1) [reply src/rviz/ogre_helpers/render_system.cpp:190]:
```
$ LANG=C apt-cache policy mesa-va-drivers 
mesa-va-drivers:
  Installed: 21.3.6~kisak1~b
  Candidate: 21.3.6~kisak1~b
  Version table:
 *** 21.3.6~kisak1~b 500
        500 http://ppa.launchpad.net/kisak/kisak-mesa/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
     20.0.8-0ubuntu1~18.04.1 500
        500 http://cz.archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages
     19.2.8-0ubuntu0~18.04.2 500
        500 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages
     18.0.0~rc5-1ubuntu1 500
        500 http://cz.archive.ubuntu.com/ubuntu bionic/universe amd64 Packages
```

Comment (peci1) [reply src/rviz/ogre_helpers/render_system.cpp:190]: That's on Bionic HWE with kisak-mesa PPA. So it seems 20.x is available via bionic-updates for everyone.

**Comment by rhaschke** [comment]
I'm not sure I fully understood your comments. Thus, let me try to summarize. @peci1 and @simonschmeisser, please correct!
- With Mesa < 21.0.3 we had a bug with PC2 display showing huge point sizes: 
  #1508
  A corresponding test is available in [`src/test/send_point_cloud_2.py`](https://github.com/ros-visualization/rviz/blob/noetic-devel/src/test/send_point_cloud_2.py)
- Mesa 21.0.3 was fixing this and it is released, both for [Bionic (via bionic-updates) and Focal (natively)](https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=libgl1-mesa-dev&searchon=names)
  Hence, it should be safe to drop the Mesa workaround.
- @simonschmeisser observed some new? issue on AMD Renoir, with PC2 display showing all styles other than points as tiny dots only, i.e. ignoring the size attribute
  Is this reproducible on other graphic cards or is it specific to AMD? Please file a new issue for this.

**Comment by peci1** [comment]
> * Mesa 21.0.3 was fixing this and it is released, both for Bionic (via bionic-updates)

Bionic has only Mesa 20. So this fix hasn't been released there. I (and probably others) are using a customized Mesa PPA to get a newer Mesa on Bionic, as that's the only way to get the new Ryzen GPUs working (together with mainline kernels).

**Comment by peci1** [comment]
I tested on Bionic+Ryzen GPU+Mesa 21 with and without the Mesa workaround:

With workaround: everything works (I haven't tested performance, though).

Without workaround: Points are displayed with correct size, but there is this (new?) issue with everything except points displaying as a single white dot:
rviz_screenshot_2022_03_07-12_35_45

I'll open an issue for the "newly discovered" bug, but I think this PR needs to be hold off until that issue is solved.

**Comment by rhaschke** [comment]
> > * Mesa 21.0.3 was fixing this and it is released, both for Bionic (via bionic-updates)
> 
> Bionic has only Mesa 20. So this fix hasn't been released there.

You are right. I was writing my comment in a rush obviously...

**Comment by peci1** [comment]
> note that you can manually enforce an older OpenGL standard: `rviz --opengl 310`.

Yes, but this combines very badly with already existing launch files starting rviz. Maybe if it were an environment variable, it would be easier to set.

**Comment by rhaschke** [comment]
What about `alias rviz='rviz --opengl 310'` in your `.bashrc`?

**Comment by peci1** [comment]
`<node name="rviz" pkg="rviz" type="rviz" args="-d path/to/config.rviz />` inside an (externally provided) launch file is the hard thing... the alias won't help there

**Comment by rhaschke** [comment]
Closing for now, as our GLSL 1.50 scripts are not Mesa-compatible yet.

**Pull request closed without merging**